Tests/SonarScanner.MSBuild.Common.Test/ILoggerTests.cs

```diff
@@ -398,4 +391,44 @@ Actual value was 100.
         var logger = Substitute.For<ILogger>();
         logger.Invoking(x => x.Log((LoggerVerbosity)100, "message")).Should().ThrowExactly<ArgumentOutOfRangeException>().WithMessage(expectedMessage);
     }
+
+    [TestMethod]
+    public void ConsoleLogger_WriteTelemetryMessages_GeneratesFile()
+    {
+        var logger = new ConsoleLogger(false, fileWrapper);
+        fileWrapper.Exists(Arg.Any<string>()).Returns(false);
+        logger.AddTelemetryMessage("key1", "value1");
+        logger.AddTelemetryMessage("key2", "value2");
+        logger.AddTelemetryMessage("key3", "value3");
+
+        const string outputDir = "outputDir";
+        var contents = new StringBuilder()
```

Comment:
This is needed so that the correct line ending is attached by the string builder depending on the OS.

Reply:
Please add a comment about this to the code.

Reply:
Move the `var contents =` declaration after the `logger.WriteTelemetry(outputDir);`. It is part of the assertions.